### PR TITLE
hotfix/outbinding: compare binding operation failed

### PR DESCRIPTION
**Pull request by 1046102779**
image
image

image

 Component-contrib repository  should provide a Equal method in OperationKind

```golang
func (o OperationKind) Equal(target OperationKind) bool {
  return strings.Equal(o, target)
}
```

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -593,7 +593,8 @@ func (a *DaprRuntime) sendToOutputBinding(name string, req *bindings.InvokeReque
 	if binding, ok := a.outputBindings[name]; ok {
 		ops := binding.Operations()
 		for _, o := range ops {
-			if o == req.Operation {
+			// ::TODO OperationKind provide a Equal method
+			if strings.EqualFold(string(o), string(req.Operation)) {
```
Comment (mukundansundar) [inline pkg/runtime/runtime.go:597]: @1046102779 as @pkedy mentioned in https://github.com/dapr/components-contrib/pull/880, most of the components have case sensitive operation names. Does this change need to be in the runtime?
I can see your example is for HTTP methods, can only the HTTP binding be changed and support both uppercase and lowercase(backwards compatibility) ? 

I am trying to understand why the comparison is changing in the runtime.

cc @pkedy

Comment (artursouza) [reply pkg/runtime/runtime.go:597]: I think it should be for HTTP component only.

**Review by artursouza — CHANGES_REQUESTED** [review]
Please, make the change in HTTP component only. We cannot guarantee that other binding components will be case insensitive.

**Comment by daixiang0** [comment]
ping @1046102779

**Pull request closed without merging**